dev-libs/libgamin/libgamin-0.1.10-r6.ebuild

```diff
@@ -27,41 +27,20 @@ RDEPEND="
 	!<app-admin/gamin-0.1.10"
 DEPEND="${RDEPEND}"
 
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.1.10-noinst-lib.patch
+	"${FILESDIR}/${PN}-0.1.10-G_CONST_RETURN-removal.patch"
+	#"${FILESDIR}/${PN}-0.1.10-crosscompile-fix.patch"
+	"${FILESDIR}/${P}-armel-features.patch"
+	"${FILESDIR}/${PN}-0.1.10-deadlock.patch"
+	"${FILESDIR}/${PN}-0.1.10-musl-pthread.patch"
+	"${FILESDIR}"/${PN}-0.1.10-compilewarnings.patch
+)
+
 src_prepare() {
 	default
 	mv "${WORKDIR}"/pkg-config-*/pkg.m4 "${WORKDIR}"/ || die
 
-	# Fix QA warnings, bug #257281, upstream #466791
-	epatch "${FILESDIR}"/${PN}-0.1.10-compilewarnings.patch
-
-	if [[ ${CHOST} != *-solaris* ]] ; then
-		# Fix compile warnings; bug #188923
-		epatch "${DISTDIR}"/gamin-0.1.9-freebsd.patch.bz2
-	else
-		# (Open)Solaris necessary patches (changes configure.in), unfortunately
-		# conflicts with freebsd patch and breaks some linux installs so it must
-		# only be applied if on solaris.
-		epatch "${DISTDIR}"/${P}-opensolaris.patch.bz2
```

Comment:
That got past me it seems, I also probably did those patches wrong, shouldn't I have just switched `epatch` with `eapply` and kept those where they were?

Reply:
`PATCHES` is better if we can as it lets us drop a phase definition, but I'd keep the order the same as they were originally (order matters for patches)

Reply:
Regarding the opensolaris patch, the format of it seems to be broken now as it can't find the files it's referencing. I've also noticed it's a dev's patch, do I fix the patch and put it in the tree or should I submit the fixes to the dev?

[build.log](https://github.com/gentoo/gentoo/files/13468314/build.log) of it failing to apply  the patch.